lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,503 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_aci
+version_added: "2.5"
+short_description: Manage an Azure Container Instance (ACI).
+description:
+    - Create, update and delete an Azure Container Instance.
+
+options:
+    resource_group:
+        description:
+            - Name of resource group.
+        required: true
+    name:
+        description:
+            - The name of the container group.
+        required: true
+        default: null
+    os_type:
+        description:
+            - The OS type of containers.
+        choices:
+            - linux
+            - windows
+        default: linux
+        required: false
+    state:
+        description:
+            - Assert the state of the ACI. Use 'present' to create or update an ACI and 'absent' to delete it.
+        default: present
+        choices:
+            - absent
+            - present
+        required: false
+    ip_address:
+        description:
+            - The IP address type of the container group.
+        choices:
+            - public
+        required: false
+        default: None
+    port:
```

Comment:
ports, and it should be a list. Actually, I wonder why we need this field, it should be a summary of the container's ports

Reply:
yes, indeed, we could just try to collect all the ports exposed by all containers and make them public, but then maybe somebody just doesn't want all the ports to be public, some may be used internally i imagine. i will change port -> ports though as it's a list.